### PR TITLE
Merging for the last time

### PR DESCRIPTION
**Pull request by freddycervantes**
This merge contains the admin feature that allows admin to set priorities for tutees.

### CODE REVIEW

**Comment by an-ju** [comment]
The four courses are hard coded.  What's the reason?

**Comment by freddycervantes** [comment]
Hey An, I'm not sure where you see class hard coded? If by four courses you mean the options for select for the question: For CS Scholars only. What CS Scholar Program Course? Those are not hard coded question, simply strings to differentiate between priorities. As stated by Chris students in the CS Scholar program for cs61a get higher priority for tutoring then students in 61b, 61c and 70. Those strings do not relate to our courses table which is used to set courses being offered for tutoring.

File: app/controllers/admins_controller.rb [hunk 1]
```diff
@@ -141,6 +145,51 @@ def post_update_password
...
   end
 
+  def update_student_priorities
+    @current_cs61a_scholars = Tutee.get_current_cs61a_sids_formatted
+    @current_cs61b_scholars = Tutee.get_current_cs61b_sids_formatted
+    @current_cs61c_scholars = Tutee.get_current_cs61c_sids_formatted
+    @current_cs70_scholars = Tutee.get_current_cs70_sids_formatted
+  end
+
+  def update_student_priorities_61A
```
Comment (an-ju) [inline app/controllers/admins_controller.rb]: This method and the following methods have course numbers.  Why is it necessary?

File: app/models/tutee.rb [hunk 2]
```diff
@@ -29,4 +29,73 @@ def validate_birth
       errors.add(:birthdate, 'Your birthdate cannot be in the future')
     end
   end
+
+  # class functions
+  class<<self
+    def get_cs61a_privileged_students
```
Comment (an-ju) [inline app/models/tutee.rb]: Same here.

File: config/routes.rb [hunk 3]
```diff
@@ -25,6 +25,12 @@
   get 'admins/update_password' => 'admins#update_password', as: :admin_update_password
   post 'admins/update_password' => 'admins#post_update_password', as: :admin_post_update_password
 
+  get 'admins/update_student_priorities' => 'admins#update_student_priorities', as: :admin_update_student_priorities
+  post 'admins/update_student_priorities_61A' => 'admins#update_student_priorities_61A', as: :admin_update_student_priorities_61A
```
Comment (an-ju) [inline config/routes.rb]: Same here.

File: app/controllers/admins_controller.rb (hunk 1 above)
Comment (tresten) [reply app/controllers/admins_controller.rb]: Priorities are stored as a string within our database. Such that a student can only have priorities ‘No’, ‘cs61a’, ‘cs61b’, ‘cs61c’, or ‘cs70’ as their priority level. This is also validated to be true in the tutee’s model. So these functions need to indicate what priority level you are trying to set a given tutee to. The hard coded courses within it are just used to find all students of the given priority level and update them accordingly.

**Pull request merged**